app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsShowMe.kt

```diff
@@ -1,74 +1,96 @@
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+private const val WOMEN = "Women"
+private const val MEN = "Men"
+private const val EVERYONE = "Everyone"
 
 /**
  * Activity to modify the show me of the User
  *
  */
+@AndroidEntryPoint
 class SettingsShowMe : AppCompatActivity() {
+
+    @Inject
+    lateinit var userHelper: UserHelper
+
+    @Inject
+    lateinit var assistedFactory: ViewModelAssistedFactory
+
+    private lateinit var viewModel: UserViewModel
+
+    private lateinit var currentShowMe: String
+    private lateinit var showMe: String
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_settings_show_me)
 
+        instantiateViewModel()
         supportActionBar?.hide()
 
-        val showMeMen = findViewById<Button>(R.id.show_me_men_button)
-        val showMeWomen = findViewById<Button>(R.id.show_me_women_button)
-        val showMeEveryone = findViewById<Button>(R.id.show_me_everyone_button)
-        val menCheckmark = findViewById<ImageView>(R.id.checkmark_img_men)
-        val womenCheckmark = findViewById<ImageView>(R.id.checkmark_img_women)
-        val everyoneCheckmark = findViewById<ImageView>(R.id.checkmark_img_everyone)
-        val checks = listOf<ImageView>(menCheckmark, womenCheckmark, everyoneCheckmark)
-        var currentShowMe = intent.getStringExtra(EXTRA_SHOW_ME)
-
-        fun setOnClickListener(button: Button, newShowMe: String, checks: List<ImageView>) {
-            button.setOnClickListener {
-                if (newShowMe != currentShowMe) {
-                    currentShowMe = newShowMe
-                    displayCheckmark(currentShowMe!!, checks)
-                }
-            }
+        showMe = intent.getStringExtra(EXTRA_SHOW_ME).toString()
+        val showMeGroup = findViewById<RadioGroup>(R.id.show_me_radio_group)
+        setClickedButton(showMeGroup, showMe)
+        val showMeWomen = findViewById<RadioButton>(R.id.women_radio_button)
+        setOnClickListener(showMeWomen, showMeGroup)
+        val showMeMen = findViewById<RadioButton>(R.id.men_radio_button)
+        setOnClickListener(showMeMen, showMeGroup)
+        val showMeEveryone = findViewById<RadioButton>(R.id.everyone_radio_button)
+        setOnClickListener(showMeEveryone, showMeGroup)
+
+        viewModel.user.observe(this) {
+            currentShowMe = it.showMe.toString()
         }
 
+    }
 
-        if (currentShowMe != null) {
-            displayCheckmark(currentShowMe!!, checks)
-            setOnClickListener(showMeMen, "Men", checks)
-            setOnClickListener(showMeWomen, "Women", checks)
-            setOnClickListener(showMeEveryone, "Everyone", checks)
+    private fun setClickedButton(button: RadioGroup, showMe: String) {
+        when (showMe) {
+            WOMEN -> button.check(R.id.women_radio_button)
+            MEN -> button.check(R.id.men_radio_button)

```

Comment:
I think that you can easily add a test for `MEN`, since you've done it for `WOMEN` and `EVERYONE`. I know it's only one line, but it will improve the coverage anyway :).